apps/portal/src/app/graphql/graphql.service.ts

```diff
@@ -15,8 +17,30 @@ export class GraphqlService {
     private authService: MsalService,
   ) {}
 
-  private getToken(): string | null {
-    return sessionStorage.getItem('accessToken');
+  private getToken(): Observable<string> {
+    const account =
+      this.authService.instance.getActiveAccount() || this.authService.instance.getAllAccounts()[0];
+
+    if (!account) {
+      // eslint-disable-next-line prefer-promise-reject-errors
+      return throwError('No active account found');
+    }
+
+    const request = {
+      scopes: [environment.apiScope],
+      account,
+    };
+    const token = this.authService.acquireTokenSilent(request).pipe(
+      switchMap((response: AuthenticationResult) => {
+        if (response && response.accessToken) {
+          return [response.accessToken];
+        }
+
+        return throwError('Failed to acquire token');
+      }),
+      catchError((error) => throwError(error)),
```

Comment:
If it cannot acquire token, do the redirect like we did in app.component.ts(you can test by intentionally failing it.)

Reply:
No it don't redirect if required i can add it